packages/devextreme/testing/tests/DevExpress.ui.widgets/selection.test.js

```diff
@@ -2442,6 +2442,58 @@ QUnit.module('onSelectionChanging', {
         assert.strictEqual(selection.isItemSelected(this.data[0]), true, 'isItemSelected returns true');
     });
 
+    QUnit.test('should be called with correct parameters when select all ', function(assert) {
+        const selectionChangingHandler = sinon.spy();
+        const dataKeys = this.data.map((item) => item.id);
+
+        this.dataSource = createDataSource(this.data, { key: 'id' }, {});
+
+        const selection = new Selection({
+            ...this.basicSelectionConfig,
+            load: () => this.dataSource.load(),
+            onSelectionChanging: selectionChangingHandler,
+        });
+
+        selection.selectAll();
+
+        assert.strictEqual(selectionChangingHandler.callCount, 1, 'selectionChanging is called once');
+        assert.deepEqual(selectionChangingHandler.getCall(0).args[0].selectedItems, this.data, 'selectedItems is correct');
+        assert.deepEqual(selectionChangingHandler.getCall(0).args[0].selectedItemKeys, dataKeys, 'selectedItemsKeys is correct');
+        assert.deepEqual(selectionChangingHandler.getCall(0).args[0].addedItemKeys, dataKeys, 'addedItemKeys is correct');
+        assert.deepEqual(selectionChangingHandler.getCall(0).args[0].addedItems, this.data, 'addedItems is correct');
+        assert.deepEqual(selectionChangingHandler.getCall(0).args[0].removedItemKeys, [], 'removedItemKeys is correct');
+        assert.deepEqual(selectionChangingHandler.getCall(0).args[0].removedItems, [], 'removedItems is correct');
+        assert.strictEqual(selectionChangingHandler.getCall(0).args[0].cancel, false, 'cancel is correct');
+        assert.ok(selectionChangingHandler.getCall(0).args[0].isSelectAll, 'isSelectAll is correct');
```

Comment:
Let's move `selectionChangingHandler.getCall(0).args[0]` to a variable to reduce copypaste and make the tests more readable?